### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by aubinleyoung**
### Blog app - integration specs for views and fix n+1 problems
In this milestone we implement the following:

- Write integration tests with Capybara gem.
- Secure app from n+1 problems.

Before solving n+1 problem,
Screenshot 2023-05-17 091256

After solving n+1 problem
Screenshot 2023-05-17 100545

### CODE REVIEW

File: spec/views/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  end
+  scenario 'display the first comment on a post' do
+    expect(page).to have_content('Hi John')
+  end
+  scenario 'display the how many comments' do
+    expect(page).to have_content("Comments: #{@first_post.comments_counter}")
+  end
+  scenario 'display the how many Likes' do
+    expect(page).to have_content("Likes: #{@first_post.likes_counter}")
+  end
```
Comment (julie-ify) [inline spec/views/post_index_spec.rb:49]: - [ ] Kindly add tests for these scenarios

  * Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
  * Tests that redirects to the post's show page when you click on a post.

File: spec/views/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    expect(page).to have_content(@user.name)
+    @first_post.comments.each do |comment|
+      expect(page).to have_content(comment.author.name)
+    end
+  end
+  scenario 'displays the comments left by each commentor' do
+    @first_post.comments.each do |comment|
+      expect(page).to have_content(comment.content)
+    end
+  end
```
Comment (julie-ify) [inline spec/views/post_show_spec.rb]: - [ ] Kindly add cover these scenarios.

  * Tests that asserts you can see who wrote the post.

File: spec/views/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    expect(page).to have_content(@user.three_recent_posts[0].title)
+    expect(page).to have_content(@user.three_recent_posts[1].title)
+    expect(page).to have_content(@user.three_recent_posts[2].title)
+    expect(page).to have_no_content(Post.where(author: @user))
+  end
+
+  it 'renders the user post index page' do
+    click_link('See all posts')
+    expect(page).to have_current_path(user_posts_path(@user))
+  end
```
Comment (julie-ify) [inline spec/views/user_show_spec.rb:53]: - [ ] Kindly cover these scenarios.

  * Tests that redirects to the post's show page when you click on a user's post.
  * Tests that redirects to the user's post's index page when you click on a **see all posts**.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @aubinleyoung, @Donmark2k and @Johnadibe,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/post_index_spec.rb (hunk 1 above)
Comment (BenMukebo) [reply spec/views/post_index_spec.rb:49]: - [ ] As requiested by the previous reviewer, Kindly make sure to add test Tests that asserts you can see a section for pagination if there are more posts than fit on the view. ✍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
### Hi @TEAM,

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

### To highlight 💪🏻
- Your Pull Request has a proper title and description. ✔️  
- Descriptive README file. ✔️  
- The app solves N+1 problem when fetching all posts and their comments for a user. :heavy_check_mark:  
 👍  Good job!


## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_


Cheers and Happy coding!👏👏👏 💻  
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by rustam-tolipov — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. No failing test !👎 ;)
4. Well-written readme 💯   

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**